app/api/discount_codes.py

```diff
@@ -23,14 +23,27 @@ class Meta:
         self_view_kwargs = {'id': '<id>'}
         inflect = dasherize
 
-    @validates_schema
-    def validate_quantity(self, data):
-        if 'max_quantity' in data and 'min_quantity' in data:
-            if data['max_quantity'] < data['min_quantity']:
+    @validates_schema(pass_original=True)
+    def validate_quantity(self, data, original_data):
+        original_data_attributes = original_data['data']['attributes']
+        discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+        if 'min_quantity' in original_data_attributes:
```

Comment:
Try something like this,

```python
if 'starts_at' not in data:
data['starts_at'] = object.starts_at
``` 
Similarly for ends_at then just compare data['starts_at'] and data['ends_at']. Simple.
Also I guess validation methods are called first before `before_update_object` so splitting the process will not work.